core/storage/email/gae_models_test.py

```diff
@@ -128,9 +153,133 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
                     sent_datetime_lower_bound='Not a datetime object')
 
 
+
+class ReplyToIdModelUnitTests(test_utils.GenericTestBase):
+
+
+    """Test the GeneralFeedbackEmailReplyToIdModel class."""
+
+    def setUp(self):
+        super(ReplyToIdModelUnitTests, self).setUp()
+
+        # pylint: disable=unused-argument
+        def _generate_unique_id_for_tests(cls):
+            return 'Non_unique_id'
+
+        self.generate_constant_id_ctx = self.swap(
+            email_models.GeneralFeedbackEmailReplyToIdModel,
+            '_generate_unique_reply_to_id',
+            types.MethodType(
+                _generate_unique_id_for_tests,
+                email_models.GeneralFeedbackEmailReplyToIdModel))
+
+    def test_id_generation_works_correctly(self):
+        # Name is too long for linter.
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        self.assertEqual(created.id, 'user_id.thread_id')
+
+        created = model.create('other_user_id', 'other_thread_id')
+        self.assertNotEqual(created.id, 'user_id.thread_id')
+
+    def test_unique_reply_id_is_unique(self):
+        # Name is too long for linter.
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        all_ids = [
+            model.create('user_id', 'thread_id').reply_to_id
+            for _ in xrange(100)]
+        set_ids = set(all_ids) #Set contains unique values
+        self.assertEqual(len(all_ids), len(set_ids))
+
+    def test_create_raises_when_duplicate_ids(self):
+        with self.generate_constant_id_ctx, self.assertRaises(Exception):
+            created1 = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+                'user_id', 'thread_id')
+            created1.put()
+            created2 = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+                'user_id', 'thread_id')
+            created2.put()
+
+    def test_get_by_reply_to_id_works_correctly(self):
+        # Name is too long for linter.
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        created.put()
+
+        result = model.get_by_reply_to_id(created.reply_to_id)
+
+        self.assertNotEqual(result, None) #There should be a result
+        self.assertEqual(result.reply_to_id, created.reply_to_id)
+
+        result = model.get_by_reply_to_id('non_existent_reply_to_id')
+
+        self.assertEqual(result, None)
+
+    def test_get_works_correctly(self):
+        created = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'thread_id')
+        created.put()
+
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+            'user_id', 'thread_id', strict=False)
+
+        self.assertNotEqual(result, 1)
```

Comment:
shouldn't it be 'assertEqual'? you can compare length of result with 1.

Reply:
Wow, I've fiddled so much with that test, I didn't see this assert made no sense. Fixed to one assert making sure result is not None and another looking for the correct id.